dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Users.java

```diff
@@ -21,6 +22,29 @@
     @Inject
     private ChangeUserIdentifierService changeUserIdentifierService;
 
+    @Inject
+    private MergeInAccountService mergeInAccountService;
+
+    @POST
+    @ApiWriteOperation
+    @Path("{consumedIdentifier}/mergeIntoUser/{baseIdentifier}")
+    public Response mergeInAuthenticatedUser(@PathParam("consumedIdentifier") String consumedIdentifier, @PathParam("baseIdentifier") String baseIdentifier) {
+
+        try {
+            User user;
+            user = findUserOrDie();
+            mergeInAccountService.mergeAccounts(user, consumedIdentifier, baseIdentifier);
+
+            return ok("All account data for " + consumedIdentifier + " has been merged into " + baseIdentifier + ".");
+        } catch (WrappedResponse ex) {
+            return ex.getResponse();
+        } catch (EJBException ise) {
+            return badRequest(ise.getCause().getMessage());
```

Comment:
I don't think there is a simple solution to the problem of wrapping exception in EJBException.
For previous task we have agreed with @madryk  to let service manage exception messages.

Reply:
I don't quite understand, if you return badRequest to the user it won't be managed by some service right?

Reply:
No it won't, but actual error message is in exception originally thrown from a service but wrapped in EJBException.
So when we catch EJBException it can be (almost) anything like: SecurityException, IllegalArgumentException ect.
We then default to get the error message from the wrapped exception rather than flatten out all different error messages being hidden by EJBException.

Reply:
OK but we still shouldn't return stacktrace to the user, he won't understand it and it is internal error reserved for developers.

Reply:
This won't be a stack trace but the original error message from bean. Example: 
![image](https://user-images.githubusercontent.com/51329176/118482792-847e5480-b715-11eb-9b19-a55088c11121.png)

Reply:
Ah ok, but what if there is going to be NullPointerException (the exceptions that we are not throwing (RuntimeExceptions)). Then we would have a situation where some exceptions possess non user friendly messages (like some exception with database).

Reply:
Plus we should also log those exceptions, otherwise the error will be lost if I'm not mistaken (maybe there is some interceptor that I don't know about).

Reply:
Afaik only RuntimeExeptions are being wrapped by EJBException. And I belive the way we handle catching Exception is descriptive enough to admin users to stay as is.

Reply:
Log entry is present and contains the original exception:

```
Caused by: java.lang.SecurityException: Only superusers can merge accounts.
	at edu.harvard.iq.dataverse.users.MergeInAccountService.mergeAccounts(MergeInAccountService.java:61)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.ejb.security.application.EJBSecurityManager.runMethod(EJBSecurityManager.java:1081)
	at org.glassfish.ejb.security.application.EJBSecurityManager.invoke(EJBSecurityManager.java:1153)
	at com.sun.ejb.containers.BaseContainer.invokeBeanMethod(BaseContainer.java:4786)
	at com.sun.ejb.EjbInvocation.invokeBeanMethod(EjbInvocation.java:656)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:822)
	at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:608)
	at edu.harvard.iq.dataverse.interceptors.LoggedCallInterceptor.callRestricted(LoggedCallInterceptor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.sun.ejb.containers.interceptors.AroundInvokeInterceptor.intercept(InterceptorManager.java:883)
	at com.sun.ejb.containers.interceptors.AroundInvokeChainImpl.invokeNext(InterceptorManager.java:822)
	at com.sun.ejb.EjbInvocation.proceed(EjbInvocation.java:608)
```

Reply:
Yes mergeAccounts is logged but findUserOrDie isn't.

Reply:
> Afaik only RuntimeExeptions are being wrapped by EJBException. And I belive the way we handle catching Exception is descriptive enough to admin users to stay as is.

I'm kinda hesitant but since there is already a mess in api which makes us not consistent anyway I guess we can leave it.